### ISSUE TITLE
`import/order` ignores `pathGroups`

### ISSUE THREAD

**Issue by TheZoker** [import-js/eslint-plugin-import#2736]
Hi there,

I'm trying to use `import/order` with this setting:
```json
"import/order": [
    "error",
    {
        "alphabetize": {
            "order": "asc",
            "caseInsensitive": false
        },
        "groups": [
            ["builtin", "external"],
            ["internal", "parent", "sibling", "index"]
        ],         
        "newlines-between": "always",
        "pathGroups": [
            {
                "pattern": "app/**",
                "group": "internal"
            }
        ],
        "distinctGroup": false
    }
],
```

When I run the code, the linter orders the imports like this:
```ts
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { ExternalUserPasswordResetModalComponent } from 'app/account/password-reset/external/external-user-password-reset-modal.component';
import { ArtemisSharedModule } from 'app/shared/shared.module';

import { accountState } from './account.route';
import { ActivateComponent } from './activate/activate.component';
```

When instead it should order it like this:
```ts
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';

import { accountState } from './account.route';
import { ActivateComponent } from './activate/activate.component';
import { ArtemisSharedModule } from 'app/shared/shared.module';
import { ExternalUserPasswordResetModalComponent } from 'app/account/password-reset/external/external-user-password-reset-modal.component';
```

Can you tell me, why my `pathGroups` and my `groups` setting to not work?

Thanks!

**Comment by inglec-arista** [comment]
I had to go digging through old issues, but the reason is that `builtin` and `external` are excluded from `pathGroups` by default:
https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md#pathgroupsexcludedimporttypes-array

You'll want to set `pathGroupsExcludedImportTypes: []`

Issue which helped:
https://github.com/import-js/eslint-plugin-import/issues/2363

**Comment by ackvf** [comment]
I found another weirdness. Using `internal` with `pathGroups` will not work, but it will work with `import/internal-regex` setting.

Before:
```js
import { X } from 'package/src/ui'
import { useCallback } from 'react'
import { Y } from 'package/src/utils'
import { something } from './something'
```
Expected:
```js
import { useCallback } from 'react'
import { X } from 'package/src/ui'
import { Y } from 'package/src/utils'
import { something } from './something'
```

This **will not** work.
```js
  rules: {
    "import/order": ["warn", {
      "groups": ["builtin", "external", "internal", "parent", "sibling", "index"],
      "pathGroups": [
        {
          "pattern": "package/**",
          "group": "internal",
        },
      ],
    }],
  },
```

This **will** work.
```js
  settings: {
    "import/internal-regex": "^package/",
  },
  rules: {
    "import/order": ["warn", {
      "groups": ["builtin", "external", "internal", "parent", "sibling", "index"],
    }],
  },
```

Why?

**Comment by akay64** [comment]
@ackvf Thank you, you saved me a lot of time. I will document this as a quirk for future reference.
